python_modules/libraries/dagster-dbt/dagster_dbt/dagster_dbt_translator.py

```diff
@@ -90,22 +90,21 @@ def settings(self) -> DagsterDbtTranslatorSettings:
 
     def get_asset_spec(
         self,
-        manifest: Mapping[str, Any],
-        dbt_nodes: Mapping[str, Any],
-        group_props: Mapping[str, Any],
-        project: Optional["DbtProject"],
-        resource_props: Mapping[str, Any],
+        manifest: "DbtManifestWrapper",
+        unique_id: str,
     ) -> AssetSpec:
         """Returns an AssetSpec representing a specific dbt resource."""
-        from dagster_dbt.dagster_dbt_translator import DbtManifestWrapper
+        # dbt_nodes = get_dbt_resource_props_by_dbt_unique_id_from_manifest(manifest)
+        group_props = {group["name"]: group for group in manifest.get_section("groups").values()}
+        resource_props = manifest.get_node(unique_id)
 
         # calculate the dependencies for the asset
-        upstream_ids = get_upstream_unique_ids(dbt_nodes, resource_props)
+        upstream_ids = get_upstream_unique_ids(manifest, resource_props)
         deps = [
             AssetDep(
-                asset=self.get_asset_key(dbt_nodes[upstream_id]),
+                asset=self.get_asset_spec(manifest, upstream_id).key,
```

Comment:
what's the quick explanation of why this isn't a perf concern / why caching was not required? specifically, seems like if `get_asset_spec` calls `get_asset_spec` on all of its upstreams, we're gonna end up re-doing work a bunch of times for assets further up in the graph.

Reply:
I think you are right that this could be an issue, I only caught this call right at the end of the day. Think i will try memoizing against `(id(manifest_json), unique_id)`